src/pymodaq_plugins_signal_recovery/daq_viewer_plugins/plugins_0D/daq_0Dviewer_Lockin_DSP7265.py

```diff
@@ -0,0 +1,187 @@
+        opts['addText'] = "Add Channel"
+        super().__init__(**opts)
+
+    def addNew(self) -> None:
+        """Add new channel to viewer
+        """
+        name_prefix = 'channel'
+
+        child_indexes = [int(par.name()[len(name_prefix) + 1:])
+                         for par in self.children()]
+
+        if child_indexes == []:
+            newindex = 0
+        else:
+            newindex = max(child_indexes) + 1
+
+        child = {
+            'title': f'Measure {newindex:02.0f}',
+            'name': f'{name_prefix}{newindex:02.0f}',
+            'type': 'itemselect',
+            'removable': True,
+            'value': dict(all_items=CHANNELS, selected=CHANNELS[0])
+        }
+
+        self.addChild(child)
+
+
+registerParameterType('dsp7265channel', ChannelGroup, override=True)
+
+
+class DAQ_0DViewer_Lockin_DSP7265(DAQ_Viewer_base):
+    """DAQ_viewer for DSP 7265 lockin
+    """
+
+    params = [
+        {'title': 'Adapter', 'name': 'adapter', 'type': 'list',
+         'limits': list(ADAPTERS.keys())},
+        {'title': 'VISA Address:', 'name': 'address', 'type': 'list',
+         'limits': VISA_RESOURCES},
+        {'title': 'ID:', 'name': 'id', 'type': 'str'},
+        {'title': 'Channels:', 'name': 'channels', 'type': 'dsp7270channel'}
+    ] + comon_parameters
+
+    def ini_attributes(self) -> None:
+        self.controller: DSP7265ThreadSafe = None
+
+    def commit_settings(self, param: Parameter) -> None:
+        """Apply the consequences of a change of value in the detector settings
+
+        Parameters
+        ----------
+        param: Parameter
+            A given parameter (within detector_settings) whose value has been
+            changed by the user
+        """
+        if param.name() in utils.iter_children(
+                self.settings.child('channels'), []):
+            data = []
+            for child in self.settings.child('channels').children():
+                labels = child.value()['selected']
+                data.append(
+                    DataFromPlugins(
+                        name=child.name(),
+                        data=[np.array([0]) for _ in labels],
+                        labels=labels,
+                        dim='Data0D'
+                    )
+                )
+            self.data_grabed_signal_temp.emit(data)
+
+    def ini_detector(self, controller: object = None) -> edict:
+        """Viewer communication initialization
+
+        Parameters
+        ----------
+        controller: (object, optional)
+            custom object of a PyMoDAQ plugin (Slave case). None if only one
+            actuator by controller (Master case)
+
+        Returns
+        -------
+        info: str
+            Id of the lockin
+        initialized: bool
+            False if initialization failed otherwise True
+        controller: object
+            Controller of the daq_viewer
+        """
+        try:
+            self.status.update(
```

Comment:
look at a recent version of the plugin template for a more concise way of doing this initialization